### PR TITLE
[Store Customization] Update patterns to use the AI selected verticals for images

### LINKED ISSUE #10623: Store Customization > Update patterns to use the AI selected verticals for images.
⚠️ This issue can be worked on as soon as https://github.com/woocommerce/woocommerce-blocks/issues/10620 is complete.

Now that all relevant information regarding the images for the patterns has been saved in the relevant wp_option (`wc_blocks_patterns_content`) in the database, the next step is to update all patterns to ensure they are populated with these images.

As described on https://github.com/woocommerce/woocommerce-blocks/issues/10503, in each one of our patterns, we can `get_option()` and then start assigning the images.

An example of how images could be populated in patterns is as follows:

```php
<img src="<?php echo isset( $image_urls[0] ) ? esc_url( $image_urls[0] ) : esc_url( plugins_url( 'images/pattern-placeholders/office.png', dirname( __FILE__ ) ) ); ?>" alt="<?php esc_attr_e( 'Placeholder image used to represent products being showcased.', 'woo-gutenberg-products-block' ); ?>" />
```

Notice how there's still a fallback to the pre-existing images in our codebase, those will be used only as a last resort, if the user doesn't have an internet connection available.

### PR DESCRIPTION
**Pull request by albarin**
## What

Update all patterns to ensure images are populated with Verticals (an API built to help designers and developers choose vertical-specific images for use in templates and automatically generated content). The displayed images should dynamically change depending on what the user decides to add as a description for their business on their AI settings screen and also during the upcoming onboarding screen.

Fixes woocommerce/woocommerce-blocks#10623

## Why

This is required to ensure all of our patterns have AI-managed images in place.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Create a new [JN install](https://jurassic.ninja/) with both Jetpack and Woocommerce installed and activated.
3. Install and activate WooCommerce Blocks from this zip file:  [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/12611550/woocommerce-gutenberg-products-block.zip)

5. In your wp-admin, you should see the following section for Jetpack:

Screenshot 2023-09-11 at 18 55 38

6. Click on "Set up Jetpack" and make sure the connection is successful:

Screenshot 2023-09-11 at 18 58 30

7. From your admin screen, click on Tools > Plugin editor: 

Screenshot 2023-09-11 at 18 34 23

8. Select the WooCommerce Blocks plugin and add the following within your woocommerce-blocks/woocommerce-gutenberg-products-block.php file and save:

```php
function connect_to_endpoint() {
	update_option( 'woocommerce_blocks_allow_ai_connection', true );

	$request = new WP_REST_Request( 'POST', '/wc/store/patterns' );

	$request->set_param( 'business_description', 'A store that sells organic food' );

	$response = rest_do_request( $request )->get_data();

	error_log( print_r( $response, true ) );
}

add_action( 'admin_init', 'connect_to_endpoint' );
```

9. Create a new post
10. Insert the Featured Category Triple pattern (or any other you want with at least one image).
11. Make sure the images on the pattern match the business description you provided: 'A store that sells organic food'
12. Go back to the plugin editor and update the business description over there to something else (e.g. books, clothes, sports gear etc.) and save
13. Return to your post editor, save it, and refresh the page.
14. Insert the same Featured Category Triple pattern again
15. Notice how all images were updated to now match what you selected as the new description for your store*

*ps: there's a limitation in terms of the number of images available in the Verticals API to populate these patterns, so not all store descriptions will necessarily return the expected images just yet.

* [x] Do not include in the Testing Notes 
* [x] Should be tested by the development team exclusively 

## Screenshots or screencast

https://github.com/woocommerce/woocommerce-blocks/assets/15730971/5f3ae0fa-483b-4c57-8833-73252372abbc

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

N/A

### CODE REVIEW

File: patterns/store-info-alt-image-and-text.php [hunk 1]
```diff
@@ -13,7 +20,7 @@
 		<div class="wp-block-column is-vertically-aligned-center" style="flex-basis:50%">
 			<!-- wp:image {"sizeSlug":"full","linkDestination":"none"} -->
 			<figure class="wp-block-image size-full">
-				<img src="<?php echo esc_url( plugins_url( 'images/pattern-placeholders/crafting-pots.png', dirname( __FILE__ ) ) ); ?>" alt="<?php esc_attr_e( 'Placeholder image used to represent a person making a clay pot.', 'woo-gutenberg-products-block' ); ?>" />
+				<img src="<?php echo esc_url( $image1 ); ?>" alt="<?php esc_attr_e( 'Placeholder image used to represent a person making a clay pot.', 'woo-gutenberg-products-block' ); ?>" />
```
Comment (samueljseay) [inline patterns/store-info-alt-image-and-text.php]: Am I understanding correctly that all these images will be generated based on store description? I'm just wondering if the image is unrelated due to store description won't the alt text be very confusing?

File: src/Patterns/PatternImages.php [hunk 2]
```diff
@@ -29,6 +29,77 @@ public function get_patterns_dictionary() {
...
 	}
 
+	/**
+	 * Returns the pattern images.
+	 *
+	 * @param string $pattern_slug The pattern slug.
+	 *
+	 * @return array The pattern images.
+	 */
+	public static function get_pattern_images( string $pattern_slug ): array {
```
Comment (samueljseay) [inline src/Patterns/PatternImages.php:39]: It feels to me like there is plenty of logic in here that would benefit from a unit test. What do you think?

File: src/Patterns/PatternImages.php [hunk 3]
```diff
@@ -29,6 +29,77 @@ public function get_patterns_dictionary() {
...
+	/**
+	 * Returns the image for the given pattern.
+	 *
+	 * @param array  $images The array of images.
+	 * @param int    $index The index of the image to return.
+	 * @param string $default_image The default image to return.
+	 *
+	 * @return string The image.
+	 */
+	public static function get_image_url( array $images, int $index, string $default_image ): string {
```
Comment (samueljseay) [inline src/Patterns/PatternImages.php:77]: this also seems like a unit test would be nice.

**Review by samueljseay — COMMENTED** [review]
Tests well, looking great to me!

I just had a couple questions around potential unit tests and also about the alt text.

File: src/Patterns/PatternImages.php (hunk 2 above)
Comment (albarin) [reply src/Patterns/PatternImages.php:39]: Totally agree, there was a `PatternImagesTest.php` with some tests for the files, but it was temporarily deleted (for more context see https://github.com/woocommerce/woocommerce-blocks/pull/10881#issuecomment-1721496102).
I'll open an issue to re-enable them and add unit tests for those two `get_image_url` and `get_pattern_images` functions, do you think that's okay?

File: src/Patterns/PatternImages.php (hunk 3 above)
Comment (albarin) [reply src/Patterns/PatternImages.php:77]: Same https://github.com/woocommerce/woocommerce-blocks/pull/10796/files#r1328597336

File: patterns/store-info-alt-image-and-text.php (hunk 1 above)
Comment (albarin) [reply patterns/store-info-alt-image-and-text.php]: It's true, now that we don't know the actual images that will be displayed, I think we should make the `alt` description more generic. I'll work on it. Thanks 🙌

**Comment by albarin** [comment]
@samueljseay thanks for your review, I've pushed some updates, could you have another look? 🙏

File: src/Patterns/PatternImages.php (hunk 2 above)
Comment (albarin) [reply src/Patterns/PatternImages.php:39]: Here's the new issue, I added it to the epic https://github.com/woocommerce/woocommerce-blocks/issues/10987

Comment (samueljseay) [reply src/Patterns/PatternImages.php:39]: Yep, no probs :)

**Review by samueljseay — APPROVED** [review]
Looks good to me thanks for updating the alt text 😄 🚀

**Pull request merged**